### ISSUE TITLE
Add_vpool failed because of 0GB writecache size

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#945]
Related to: http://tickets.openvstorage.com/browse/OVSSUP-115

The `add_vpool` call was executed via the OVS GUI.

```
2016-09-22 16:39:32 07400 +0200 - stor-07.be-g8-4 - 19232/140107787003712 - celery/celery.worker.job - 1214 - ERROR - Task ovs.storagerouter.add_vpool[a1cace67-36ea-4227-aef1-521879e9af22] raised unexpected: RuntimeError('StorageDriver service failed to start (service not running)',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 993, in add_vpool
    raise RuntimeError('StorageDriver service failed to start (service not running)')
RuntimeError: StorageDriver service failed to start (service not running)
```

```
2016-09-22 16:39:30 901333 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/SCOCache - 0000000000000015 - error - initMountPoints_: failed to initialize mountpoint "/mnt/ssd1/data01_write_sco_1": invalid capacity specified /mnt/ssd1/data01_write_sco_1 Invalid argument (22) - proceeding without this mountpoint
2016-09-22 16:39:30 901382 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/SCOCache - 0000000000000016 - fatal - initMountPoints_: no usable mountpoints available, giving up
2016-09-22 16:39:30 901976 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/VolManager - 0000000000000017 - error - VolManager: Exception during VolManager construction: no sco cache mountpoints available 
2016-09-22 16:39:30 902065 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/VolManager - 0000000000000018 - error - start: Problem creating VolManager: no sco cache mountpoints available 
2016-09-22 16:39:30 902198 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/EventPublisher - 0000000000000019 - info - ~EventPublisher: Stopping
2016-09-22 16:39:30 908503 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/volumedriver_fs - 000000000000001a - fatal - operator(): caught exception: no sco cache mountpoints available 
2016-09-22 16:39:30 908557 +0200 - stor-02.be-g8-4 - 16755/0x00007faf3fb15940 - volumedriverfs/volumedriver_fs - 000000000000001b - notice - at_exit: Time spent executing user instructions (seconds): 0.16000
```

Volumedriver settings:

```
        "scocache_mount_points": [
            {
                "path": "/mnt/ssd1/data01_write_sco_1", 
                "size": "0KiB"
            }
```

CHECKED ON TESTCLUSTER:

```
        "scocache_mount_points": [
            {
                "path": "/mnt/hdd4/vm-vpool_write_sco_1", 
                "size": "7380992KiB"
            }
```

Checking the workers for parameters:

```
2016-09-22 16:39:03 59000 +0200 - stor-07.be-g8-4 - 19232/140107787003712 - celery/celery.worker.strategy - 1211 - INFO - Received task: ovs.storagerouter.add_vpool[a1cace67-36ea-4227-aef1-521879e9af22]
2016-09-22 16:39:03 59100 +0200 - stor-07.be-g8-4 - 19232/140107787003712 - celery/celery.pool - 1212 - DEBUG - TaskPool: Apply <function _fast_trace_task at 0x7f6d5e4e8050> (args:('ovs.storagerouter.add_vpool', 'a1cace67-36ea-4227
-aef1-521879e9af22', ({u'vpool_name': u'data01', u'fragment_cache_on_write': False, u'storage_ip': u'10.109.2.42', u'readcache_size': 0, u'storagerouter_ip': u'10.109.2.42', u'config_params': {u'write_buffer': 2048, u'sco_size': 32
, u'cache_strategy': u'none', u'dedupe_mode': u'non_dedupe', u'cluster_size': 4, u'dtl_transport': u'tcp', u'dtl_mode': u'a_sync'}, u'writecache_size': 0, u'backend_connection_info': {u'username': u'a4028dbc-050b-4ab5-af2b-7c2d87ca
4ee7', u'host': u'10.109.2.48', u'password': u"?M}:TtQ.g0'5SiHGZXAwu8<#IHk_ZaHpv[p#07;h-[>8TBty!DF;!cnB6KCk]@ZJj8mpld;[1<wYs=s*}P.<=GWpp8QuSm8|_6?vCl=L+*'#cukIQcP7lt9IE.Spt*zB", 'local': True, u'port': 443, u'backend': {u'backend':
 u'fabee6a7-b870-469d-be43-1ffbe1b6203f', u'metadata': u'gigpreset'}}, u'fragment_cache_on_read': False, u'type': u'alba'},), {}, {'utc': True, u'is_eager': False, 'chord': None, u'group': None, 'args': ({u'vpool_name': u'data01', 
u'fragment_cache_on_write': False,... kwargs:{})
```

Seeing this: `u'writecache_size': 0`

**Comment by kinvaris** [comment]
Saw it a second time, with same problems

**Comment by hofkensj** [comment]
@kinvaris will this significantly impact operations in the field ?

**Comment by kinvaris** [comment]
@hofkensj: if it occurs, not really. The vPool can just be deleted ...

**Comment by hofkensj** [comment]
We agreed to make sure documentation of the known issue is there

then we can close

**Comment by khenderick** [comment]
Fixed by @kvanhijf as side effect from #852 and was packaged in openvstorage-2.7.5-rev.4281.6612d8c
